packages/builder/src/services/pinata.ts

```diff
@@ -1,42 +1,48 @@
-// eslint-disable-next-line
-const JWT = process.env.REACT_APP_PINATA_JWT;
-// eslint-disable-next-line
-const GATEWAY = process.env.REACT_APP_PINATA_GATEWAY;
-const PIN_JSON_TO_IPFS_URL = "https://api.pinata.cloud/pinning/pinJSONToIPFS";
-const PIN_FILE_TO_IPFS_URL = "https://api.pinata.cloud/pinning/pinFileToIPFS";
+import { z } from "zod";
 
 export default class PinataClient {
   private jwt: string;
 
   private gateway: string;
 
-  private pinJSONToIPFSURL: string;
+  private pinataBaseUrl: string;
 
-  private pinFileToIPFSURL: string;
+  private pinJSONToIPFSUrl: string;
+
+  private pinFileToIPFSUrl: string;
 
   constructor() {
-    if (JWT === undefined || GATEWAY === undefined) {
-      throw new Error(
-        "remember to set the REACT_APP_PINATA_JWT and REACT_APP_PINATA_GATEWAY env vars"
-      );
-    }
-    this.jwt = JWT!;
-    this.gateway = GATEWAY.replace(/\/$/, "");
-    this.pinJSONToIPFSURL = PIN_JSON_TO_IPFS_URL;
-    this.pinFileToIPFSURL = PIN_FILE_TO_IPFS_URL;
+    this.jwt = z
+      .string({
+        required_error: "REACT_APP_PINATA_JWT is required",
+      })
+      .parse(process.env.REACT_APP_PINATA_JWT);
+    this.gateway = z
+      .string({ required_error: "REACT_APP_IPFS_BASE_URL is required" })
+      .url()
+      .parse(process.env.REACT_APP_IPFS_BASE_URL)
+      .replace(/\/$/, "");
+    this.pinataBaseUrl = z
+      .string({ required_error: "REACT_APP_PINATA_BASE_URL" })
+      .url()
+      .parse(process.env.REACT_APP_PINATA_BASE_URL)
+      .replace(/\/$/, "");
```

Comment:
Not in scope for this PR, but it would be nice in the future to decouple environment access from object instantiation, so that the object can be created in more contexts (notably, tests) without mocking an environment around it. So something like:

```
const config = readConfigFromEnv();
// ...
const pinataClient = new PinataClient({
  jwt: config.pinataJwt,
  baseUrl: config.pinataIpfsBaseUrl
})
```

Reply:
I created a getEnv function in common, for now the class is still initializing them in the constructor but now we can mock getEnv in tests, wdyt?

Reply:
Again I don't think it's necessary to address this in the PR but just to explain where I'm coming from, my reasoning is that if you make a class read configuration directly from the environment, the way to configure it is decided by the class author; if you make it read configuration from constructor arguments, they way to configure it is decided by the class user. The latter is more future-proof. Workarounds are possible when you can hook into the module loader (like with module mocks during testing) but even there it to me it looks like two wrong trying to make a right, when  the right was low-hanging to begin with.